base/ca/src/main/java/com/netscape/cms/servlet/cert/ListCerts.java

```diff
@@ -141,10 +140,8 @@ public void init(ServletConfig sc) throws ServletException {
         }
         if (sc.getInitParameter(ALLOWED_CLIENT_FILTERS) == null
                 || sc.getInitParameter(ALLOWED_CLIENT_FILTERS).equals("")) {
-            mAllowedClientFilters.addElement("(certStatus=*)");
-            mAllowedClientFilters.addElement("(certStatus=VALID)");
-            mAllowedClientFilters.addElement("(|(certStatus=VALID)(certStatus=INVALID)(certStatus=EXPIRED))");
-            mAllowedClientFilters.addElement("(|(certStatus=VALID)(certStatus=REVOKED))");
+            mAllowedClientFilters.addElement("(\\\\(\\\\&)?(\\\\(\\\\|)?(\\\\(certStatus=(\\\\*|VALID|INVALID|EXPIRED)\\\\))*(\\\\))?(\\\\(certRecordId(<|>)=(0x)?\\\\d+\\\\))*(\\\\))?");
+            mUseClientFilterRegexp = true;
```

Comment:
I'm a bit concerned about using regex here since it's not easy to read and might be difficult to change in case we need to add more filters. If we're going to keep the regex I think we need to include some explanation what filters will match the regex.

Reply:
OK, I'll add the comment. Since the filter now include the cert serial the previous approach becomes awkward.

Reply:
Could you also include the regex in the comment without the escape characters? That probably will make it easier to see how those filters can be captured by the regex.